src/SourceBrowser.Generator/Transformers/TreeViewTransformer.cs

```diff
@@ -34,20 +35,55 @@ public TreeViewTransformer(string savePath, string userName, string repoName)
 
         protected override void VisitWorkspace(WorkspaceModel workspaceModel)
         {
-            using (var stringWriter = new StreamWriter(_savePath, false))
-            using(_writer = new HtmlTextWriter(stringWriter))
+            // The first WorkspaceModel that is visited is the root of the tree view
+            // and its children are the solutions.
+
+            bool disposeWriters = false;
+
+            // Create the writers only if they're null
+            if (_sw == null)
             {
+                _sw = new StreamWriter(_savePath, false);
+
+                if (_writer != null)
+                    _writer.Dispose();
+
+                _writer = new HtmlTextWriter(_sw);
+
+                disposeWriters = true;
+            }
+
+            if (disposeWriters)
+            {
+                // The current WorkspaceModel is the root node, no need to increase the depth
                 _writer.AddAttribute(HtmlTextWriterAttribute.Id, "browserTree");
                 _writer.AddAttribute(HtmlTextWriterAttribute.Class, "treeview");
                 _writer.AddAttribute("data-role", "treeview");
                 _writer.RenderBeginTag(HtmlTextWriterTag.Ul);
-
+            }
+            else
+            {
+                // The current WorkspaceModel is a Child of the root node.
                 depth++;
-                base.VisitWorkspace(workspaceModel);
-                depth--;
+            }
+            
+            base.VisitWorkspace(workspaceModel);
 
+            if (disposeWriters)
+            {
+                // The current WorkspaceModel is the root node.
+                // Every child has been visited: dispose the writers.
+
+                disposeWriters = false;
                 _writer.RenderEndTag();
                 _writer.WriteLine();
+                _writer.Dispose();
+                _sw.Dispose();
```

Comment:
I'll address here your doubts about the changes in the `TreeViewTransformer.VisitWorkspace` method: 

the changes that I made to make the application able to process multiple solutions in the same repository were to process all the .sln files in parallel and when this processing was finished a *new* WorkspaceModel item was created and it's children were the WorkspaceModel *resulted* from the solutions.

When the TreeViewTransformer visited the first WorkspaceModel (which is the root) it created the writers and then it invoked the base method, which recursively visited all the sub nodes.
This means that when the transformer visited the solutions' WorkspaceModels under the root, it created again the writers (because `projectItem` is a WorkspaceModel), and therefore tried to create the same output file, which lead to an IOException: "The process cannot access the file 'treeView.html' because it is being used by another process."

Anyway, what I wrote was more of a hack than a proper fix, so I'd like to use another approach: by restoring the TreeViewTransformer class as it was before of my changes, we can bypass the same problem by adding all the solutions' children to the root node, instead of the solutions' WorkspaceModels.

So, in the `UploadRepository.ProcessRepo` method, replace
```csharp
// Add all the results to the root workspace model.
foreach (var workspace in processedWorkspaces)
    rootWorkspaceModel.Children.Add(workspace);

// WorkspaceModel (root)
// - WorkspaceModel (sln1)
//   - FolderModel
//     - ...
// - WorkspaceModel (sln2)
//   - FolderModel
//     - ...
```
with
```csharp
foreach (var children in processedWorkspaces.SelectMany(e => e.Children))
    rootWorkspaceModel.Children.Add(children);

// WorkspaceModel (root)
// - FolderModel (from sln1's WorkspaceModel)
//   - ...
// - FolderModel (from sln2's WorkspaceModel)
//   - ...
```
resulting in a single `WorkspaceModel`.

Please, give me your opinions about this. Thanks!